BParkingNano/plugins/TrackMerger.cc

```diff
@@ -151,6 +166,61 @@ void TrackMerger::produce(edm::StreamID, edm::Event &evt, edm::EventSetup const
     float DCASig = (DCABSErr != 0 && float(DCABSErr) == DCABSErr) ? fabs(DCABS/DCABSErr) : -1;
     if (DCASig >  dcaSig_  && dcaSig_ >0) continue;
 
+    // clean tracks wrt to all muons
+    int matchedToMuon       = 0;
+    int matchedToLooseMuon  = 0;
+    int matchedToSoftMuon   = 0;
+    int matchedToMediumMuon = 0;
+    for (const pat::Muon &imutmp : *muons) {
+        for (unsigned int i = 0; i < imutmp.numberOfSourceCandidatePtrs(); ++i) {
+            
+            const edm::Ptr<reco::Candidate> & source = imutmp.sourceCandidatePtr(i);
+            if (! ((imutmp.sourceCandidatePtr(i)).isNonnull() && 
+                   (imutmp.sourceCandidatePtr(i)).isAvailable())
+               )   continue;
+
+            const reco::Candidate & cand = *(source);
+            if (cand.charge() == 0 || cand.bestTrack() == nullptr) continue;
+            try { cand.bestTrack()->eta(); }
```

Comment:
Why this try-catch?

Reply:
It happened to me (not in this framework but the piece of code was the same) that the pointer wasn't null but then for some reason the code was crashing at runtime.
Here I checked w/o it on 1k events and nothing bad happened, but I kept the try/catch anyhow since 1k is not very large..